src/core/main.go

```diff
@@ -208,6 +208,25 @@ func main() {
 		log.Fatalf("Failed to initialize API handlers with error: %s", err.Error())
 	}
 
+	if config.WithTrivy() {
```

Comment:
@danielpacak @heww 

As now we may have more than one default scanners installed, I'd like to suggest refactoring the `EnsureScanner ` and `RemoveImmutableScanner` to support multiple scanners.

After that, this logic may be changed to :

```
scanners:=make([]*scanner.Registration, 0)

if config.WithTrivy(){
  scanners = append(scanners, &scanner.Registration{Trivy})
}

if config.WithClair() {
  scanners = append(scanners, &scanner.Registration{Clair})
}

if err := scan.EnsureScanner(scanners, true); err != nil {
   log.Fatalf("failed to register Trivy scanner: %v", err)
}

// We can also make sure the case that if both two are installed, Trivy will be the default

// For the scanner not in the scanners list, remove them

scan.RemoveImmutableScanner(notInsatlledScanners)

```

Reply:
> @danielpacak @heww
> 
> As now we may have more than one default scanners installed, I'd like to suggest refactoring the `EnsureScanner ` and `RemoveImmutableScanner` to support multiple scanners.
> 
> After that, this logic may be changed to :
> 
> ```
> scanners:=make([]*scanner.Registration, 0)
> 
> if config.WithTrivy(){
>   scanners = append(scanners, &scanner.Registration{Trivy})
> }
> 
> if config.WithClair() {
>   scanners = append(scanners, &scanner.Registration{Clair})
> }
> 
> if err := scan.EnsureScanner(scanners, true); err != nil {
>    log.Fatalf("failed to register Trivy scanner: %v", err)
> }
> 
> // We can also make sure the case that if both two are installed, Trivy will be the default is covered
> 
> // For the scanner not in the scanners list, remove them
> 
> scan.RemoveImmutableScanner(notInsatlledScanners)
> ```

@hewe WDYT? I'm okey with such refactoring. We could also take it as smaller follow up PR as there're unit tests to be fixed / adjusted

Reply:
@steven-zou @heww please review the refactored code for registering scanners